pytorch_lightning/plugins/checkpoint/torch.py

```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.checkpoint.checkpoint import CheckpointIOPlugin
+from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.utilities.cloud_io import atomic_save
+from pytorch_lightning.utilities.cloud_io import load as pl_load
+
+
+class TorchCheckpointIOPlugin(CheckpointIOPlugin):
+    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
+        # dump states as a checkpoint dictionary object
+        try:
+            # write the checkpoint dictionary on the file
+            atomic_save(checkpoint, filepath)
+        except AttributeError as err:
+            key = pl.LightningModule.CHECKPOINT_HYPER_PARAMS_KEY
+            checkpoint.pop(key, None)
+            rank_zero_warn(f"Warning, `{key}` dropped from checkpoint. An attribute is not picklable: {err}")
+            atomic_save(checkpoint, filepath)
```

Comment:
n00b question: is there more context for this try/catch?

Reply:
Absolutely no idea, this block of code has remained the same whilst being moved across the codebase

Reply:
I went really really deep into the weeds and I think I found the source @ananthsub 

https://github.com/PyTorchLightning/pytorch-lightning/pull/431

So from what I see it's around saving objects that are not pickle-able?